Framework/DataHandling/src/LoadPSIMuonBin.cpp

```diff
@@ -356,185 +472,304 @@ void LoadPSIMuonBin::assignOutputWorkspaceParticulars(
+    addToSampleLog("Mon High", "0", outputWorkspace);
   } else {
     for (auto i = 0u; i < 4; ++i) {
       if (m_header.monLow[i] == 0 || m_header.monHigh[i] == 0)
         break;
-      logAlg->setProperty("LogType", "String");
-      logAlg->setProperty("LogName", "Mon Low" + std::to_string(i));
-      logAlg->setProperty("LogText", std::to_string(m_header.monLow[i]));
-      logAlg->executeAsChildAlg();
-
-      logAlg->setProperty("LogType", "String");
-      logAlg->setProperty("LogName", "Mon High + i");
-      logAlg->setProperty("LogText", std::to_string(m_header.monHigh[i]));
-      logAlg->executeAsChildAlg();
+      addToSampleLog("Mon Low " + std::to_string(i), m_header.monLow[i],
+                     outputWorkspace);
+      addToSampleLog("Mon High" + std::to_string(i), m_header.monHigh[i],
+                     outputWorkspace);
     }
   }
 
-  // Mon Deviation
-  logAlg->setProperty("LogType", "String");
-  logAlg->setProperty("LogName", "Mon Deviation");
-  logAlg->setProperty("LogText", m_header.monDeviation);
-  logAlg->executeAsChildAlg();
+  addToSampleLog("Mon Deviation", m_header.monDeviation, outputWorkspace);
 
   if (m_header.realT0[0] != 0) {
-    for (const float &i : m_header.realT0) {
-      if (i == 0)
+    // 16 is the max size of realT0
+    for (auto i = 0u; i < 16; ++i) {
+      if (m_header.realT0[i] == 0)
         break;
-      logAlg->setProperty("LogType", "String");
-      logAlg->setProperty("LogName", "realT0 + i");
-      logAlg->setProperty("LogText", std::to_string(i));
+      addToSampleLog("realT0 " + std::to_string(i), m_header.realT0[i],
+                     outputWorkspace);
+    }
+  }
+
+  // Read in the temperature file if provided/found
+  try {
+    readInTemperatureFile(outputWorkspace);
+  } catch (std::invalid_argument &e) {
+    g_log.warning("Temperature file could not be loaded: " +
+                  std::string(e.what()));
+  } catch (std::runtime_error &e) {
+    g_log.warning("Temperature file could not be loaded:" +
+                  std::string(e.what()));
+  }
+}
+
+void LoadPSIMuonBin::processTitleHeaderLine(const std::string &line) {
+  bool titlesFound = false;
+  std::string foundTitles = "";
+  for (const auto &charecter : line) {
+    if (charecter == ':') {
+      titlesFound = true;
+      continue;
+    } else if (titlesFound) {
+      foundTitles += charecter;
+    }
+  }
+  boost::trim(foundTitles);
+  if (foundTitles.find("\\") == std::string::npos) {
+    boost::split(m_tempHeader.titles, foundTitles, boost::is_any_of(" "));
+    m_tempHeader.delimeterOfTitlesIsBackSlash = false;
+  } else {
+    boost::split(m_tempHeader.titles, foundTitles, boost::is_any_of("\\"));
+    m_tempHeader.delimeterOfTitlesIsBackSlash = true;
+  }
+} // namespace DataHandling
+
+void LoadPSIMuonBin::processDateHeaderLine(const std::string &line) {
+  // Assume the date is added in the same place as always
+  auto date = line.substr(2, 11);
```

Comment:
can we have a comment explaining why the numbers are 2, 11, 14 and 8?

Reply:
My use of magic numbers in this algorithm is based on the assumption of where parts of the file will be at all times. I will add comments.